File: src/ServerPulse.Frontend/src/app/modules/authentication/domain/dtos/confirmEmailAccessTokenData.ts
import { AccessTokenData } from "../..";

export interface ConfirmEmailAccessTokenData {
    accessToken: string;
    refreshToken: string;
    refreshTokenExpiryDate: Date;
}

export function mapConfirmEmailAccessTokenDataToAuthToken(response: ConfirmEmailAccessTokenData): AccessTokenData {
    return {
        accessToken: response?.accessToken,
        refreshToken: response?.refreshToken,
        refreshTokenExpiryDate: new Date(response?.refreshTokenExpiryDate)
    }
}

File: src/ServerPulse.Backend/AuthenticationApi.Infrastructure/Dtos/Endpoints/Auth/ConfirmEmail/ConfirmEmailAccessTokenData.cs
namespace AuthenticationApi.Infrastructure.Dtos.Endpoints.Auth.ConfirmEmail
{
    public class ConfirmEmailAccessTokenData
    {
        public string AccessToken { get; set; } = string.Empty;
        public string RefreshToken { get; set; } = string.Empty;
        public DateTime? RefreshTokenExpiryDate { get; set; }
    }
}
